tasks/internal/push-artifacts-to-cdn-task/push-artifacts-to-cdn-task.yaml

```diff
@@ -773,6 +779,85 @@ spec:
+          value: "$(params.snapshot_json)"
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        STDERR_FILE=/tmp/stderr.txt
+
+        # Check if the previous step finished successfully. If not, stop here.
+        if [ "$(cat "$(results.result.path)")" != "Success" ]; then
+          echo "Previous step failed. Exiting..."
+          exit 0
+        fi
+
+        exitfunc() {
+            local err=$1
+            local line=$2
+            local command="$3"
+            if [ "$err" -eq 0 ] ; then
+                echo -n "Success" > "$(results.result.path)"
+            else
+                echo "$0: ERROR '$command' failed at line $line - exited with status $err" > "$(results.result.path)"
+                if [ -f "$STDERR_FILE" ] ; then
+                    # save output to the task result excluding the trace log
+                    tail -n 20 "$STDERR_FILE" |grep -v "^\+"| tee -a "$(results.result.path)"
+                fi
+            fi
+            exit 0 # exit the script cleanly as there is no point in proceeding past an error or exit call
+        }
+        # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        CONTENT_DIR=/shared/artifacts
+        SIGNED_DIR="$CONTENT_DIR/signed"
+        COMPRESSED_DIR="$CONTENT_DIR/compressed"
+
+        function compress_components {
+
+          mkdir -p "$COMPRESSED_DIR"
+          COMPONENT="$1"
+          NUM_MAPPED_FILES=$(jq '.staged.files | length' <<< "${COMPONENT}")
+          for ((i = 0; i < NUM_MAPPED_FILES; i++)) ; do
+            FILE=$(jq -c --arg i "$i" '.staged.files[$i|tonumber]' <<< "$COMPONENT")
+            SOURCE=$(jq -er '.source' <<< "$FILE") \
+              || (echo "Missing staged.files.source for component." && exit 1 )
+
+            case "${SOURCE}" in
+              (*darwin*)
+                DARWIN_FILE=$(find "${SIGNED_DIR}" -type f -name \*darwin\*)
+                tar -czf "${SIGNED_DIR}/${SOURCE}" -r "${DARWIN_FILE}"
+                rm -f "${DARWIN_FILE}"
+                ;;
+              (*linux*)
+                LINUX_FILE=$(find "${SIGNED_DIR}" -maxdepth 1 -type f -name \*linux\*)
+                tar -czf "${SIGNED_DIR}/${SOURCE}" -r "${LINUX_FILE}"
+                rm -f "${LINUX_FILE}"
+                ;;
+              (*windows*)
+                WINDOWS_FILE=$(find "${SIGNED_DIR}" -type f -name \*windows\*)
+                zip -r "${SIGNED_DIR}/${SOURCE}" "${WINDOWS_FILE}"
+                rm -f "${WINDOWS_FILE}"
+                ;;
+            esac
+          done
+        }
+
+        NUM_COMPONENTS=$(jq '.components | length' <<< "$SNAPSHOT_JSON")
+        for ((i = 0; i < NUM_COMPONENTS; i++)) ; do
+            COMPONENT=$(jq -c --arg i "$i" '.components[$i|tonumber]' <<< "$SNAPSHOT_JSON")
+            # eventually the function does not redirect the error to stderr, so looking for all output
```

Comment:
What does this mean? I don't understand.

Reply:
this comment is a left over. Nice you spot, removing.

Reply:
removed.